### PR TITLE
Add composer 2.4 image

### PR DESCRIPTION
**Pull request by mateuszsip**
Hi!
Composer 2.4 is released and includes very useful `audit` command, so it would be great to start publishing 2.4 images

https://blog.packagist.com/composer-2-4/

Closes https://github.com/composer/docker/issues/256

### CODE REVIEW

File: .github/workflows/v2.yaml [hunk 1]
```diff
@@ -26,8 +27,6 @@ jobs:
         docker build \
         --pull \
         --no-cache \
-        --tag composer/composer:latest \
```
Comment (mateuszsip) [inline .github/workflows/v2.yaml:29]: I think it doesn't make sense to push those tags for non-latest versions

Comment (alcohol) [reply .github/workflows/v2.yaml:29]: You're right, it does not.

File: .github/workflows/v2.yaml [hunk 2]
```diff
@@ -72,8 +65,6 @@ jobs:
...
-        --tag composer/composer:2 \
         --tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+(-RC\d)?' Dockerfile | grep -oP '2.3.\d+(-RC\d)?') \
```
Comment (alcohol) [inline .github/workflows/v2.yaml:68]:
```suggestion
        --tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+' Dockerfile | grep -oP '2.3.\d+') \
```

File: .github/workflows/v2.yaml [hunk 3]
```diff
@@ -85,8 +76,6 @@ jobs:
     - name: Push tag(s) to Docker Hub
       if: github.ref == 'refs/heads/main'
       run: |
-        docker push composer/composer:latest
-        docker push composer/composer:2
         docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+(-RC\d)?' Dockerfile | grep -oP '2.3.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:79]:
```suggestion
        docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+' Dockerfile | grep -oP '2.3.\d+')
```

File: .github/workflows/v2.yaml [hunk 4]
```diff
@@ -98,9 +87,51 @@ jobs:
     - name: Push tag(s) to Amazon Public ECR
       if: github.ref == 'refs/heads/main'
       run: |
-        docker tag composer/composer:latest $ECR_REPO:latest
-        docker tag composer/composer:2 $ECR_REPO:2
         docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+(-RC\d)?' Dockerfile | grep -oP '2.3.\d+(-RC\d)?') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.3.\d+(-RC\d)?' Dockerfile | grep -oP '2.3.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:90]:
```suggestion
        docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.3.\d+' Dockerfile | grep -oP '2.3.\d+') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.3.\d+' Dockerfile | grep -oP '2.3.\d+')
```

File: .github/workflows/v2.yaml [hunk 5]
```diff
@@ -98,9 +87,51 @@ jobs:
...
-        docker push $ECR_REPO:latest
-        docker push $ECR_REPO:2
         docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.3.\d+(-RC\d)?' Dockerfile | grep -oP '2.3.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:91]:
```suggestion
        docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.3.\d+' Dockerfile | grep -oP '2.3.\d+')
```

File: .github/workflows/v2.yaml [hunk 6]
```diff
@@ -98,9 +87,51 @@ jobs:
...
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Push tag(s) to Docker Hub
+        if: github.ref == 'refs/heads/main'
+        run: |
+          docker push composer/composer:latest
+          docker push composer/composer:2
+          docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.4.\d+(-RC\d)?' Dockerfile | grep -oP '2.4.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:121]:
```suggestion
          docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.4.\d+' Dockerfile | grep -oP '2.4.\d+?')
```

File: .github/workflows/v2.yaml [hunk 7]
```diff
@@ -98,9 +87,51 @@ jobs:
...
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.AWS_ECR_ACCESS_KEY }}
+          password: ${{ secrets.AWS_ECR_SECRET_KEY }}
+      - name: Push tag(s) to Amazon Public ECR
+        if: github.ref == 'refs/heads/main'
+        run: |
+          docker tag composer/composer:latest $ECR_REPO:latest
+          docker tag composer/composer:2 $ECR_REPO:2
+          docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.4.\d+(-RC\d)?' Dockerfile | grep -oP '2.4.\d+(-RC\d)?') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.4.\d+(-RC\d)?' Dockerfile | grep -oP '2.4.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:134]:
```suggestion
          docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.4.\d+' Dockerfile | grep -oP '2.4.\d+') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.4.\d+' Dockerfile | grep -oP '2.4.\d+')
```

File: .github/workflows/v2.yaml [hunk 8]
```diff
@@ -98,9 +87,51 @@ jobs:
...
+          docker push $ECR_REPO:latest
+          docker push $ECR_REPO:2
+          docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.4.\d+(-RC\d)?' Dockerfile | grep -oP '2.4.\d+(-RC\d)?')
```
Comment (alcohol) [inline .github/workflows/v2.yaml:137]:
```suggestion
          docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.4.\d+' Dockerfile | grep -oP '2.4.\d+?')
```

**Comment by alcohol** [comment]
Thanks!

**Pull request merged**